core/src/test/java/org/apache/calcite/test/JdbcTest.java

```diff
@@ -8056,6 +8056,17 @@ private void checkGetTimestamp(Connection con) throws SQLException {
         .returns("C1=OBJECT; C2=ARRAY; C3=INTEGER; C4=BOOLEAN\n");
   }
 
+  @Test void testJsonQuery() {
```

Comment:
how about some negative tests, which are supposed to fail?

Reply:
I added negative tests for the cases I introduced.